### PR TITLE
Decorate class

### PR DESCRIPTION
**Pull request by MustakimMasum**
### Activities of this branch

- Added nameable class
- Added base_decorator class
- Added capitalize_decorator class
- Added timmer_decorator class
- Tested the decoration

Kindly asking for a review

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
```
Comment (MahmoudBakr23) [inline base_decorator.rb:6]: - [ ] Please make sure to add the association between the Namebale and BaseDecorator classes to ensure correct readability and accessibility between the two classes. 🙏 
`attr_accessor :nameable`

File: trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require_relative 'base_decorator'
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.slice(0, 10)
```
Comment (MahmoudBakr23) [inline trimmer_decorator.rb]: - [ ] This logic of trimming can be modified by a condition function, so basically, we want to tell the class to only trim the name if it exceeds 10 characters. Please implement the correct logic for this. 👍

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @firepicaso 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done working on the Decorator classes so far with the custom correct_name methods
- Descriptive PR and README
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OLIPLICHE — APPROVED** [review]
Hi @firepicaso, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Applied the changes requested by the previous review 👏 👏
-  Implemented decorator (✅)) 👏
- README file is descriptive 👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**